pkg/query-service/app/http_handler.go

```diff
@@ -793,6 +794,54 @@ func (aH *APIHandler) queryDashboardVars(w http.ResponseWriter, r *http.Request)
 	aH.Respond(w, dashboardVars)
 }
 
+func prepareQuery(r *http.Request) (string, error) {
+	var postData *model.DashboardVars
+
+	if err := json.NewDecoder(r.Body).Decode(&postData); err != nil {
+		return "", fmt.Errorf("failed to decode request body: %v", err)
+	}
+
+	query := strings.TrimSpace(postData.Query)
+
+	if query == "" {
+		return "", fmt.Errorf("query is required")
+	}
+	if strings.Contains(strings.ToLower(query), "alter table") {
```

Comment:
Should we also check from truncate and drop table ?

Reply:
Sure, let me add all the mutation kinds here.

Reply:
Yeah, or can create a separate file for checking this, can be utilised by other functions as well.

Reply:
This is a very basic check against the queries run. I am not sure if it is worth being shared. Ideally, the user should be limited to read privileges here. I believe I will keep limited to this for now and create a new issue for read only user.